### DISCUSSION TITLE
Recently added features

### DISCUSSION THREAD

**Discussion by reyis4444** [Divested-Mobile/DivestOS-Build#70] [Q&A]
18.1 now provides an option to block secondary users from installing apps.
18.1 now allows prohibiting native debugging via ptrace_scope thanks to patches from GrapheneOS.
18.1 now features timeouts to automatically reboot, disable Bluetooth, and disable Wi-Fi thanks to patches from GrapheneOS and CalyxOS.

Hi, These features are what I've been waiting for for a long time for DivestOS. But my supported device is running Android 17.1.

Does it intend to provide these features in DivestOS 17.1 and below?

**Comment by SkewedZeppelin** [comment]
I already mostly ported the timeouts to 17.1.
The other two should be doable too.
I won't put an eta on it.

**Reply by SkewedZeppelin** [reply]
The timeouts and ptrace scope option is now available on 17.1, https://github.com/Divested-Mobile/DivestOS-Build/commit/a53062ca0ba8ea90dfbc219e072c662fdb959cb8

**Comment by rsssxd** [comment]
Which versions have exec-spawning?

**Reply by SkewedZeppelin** [reply]
exec_spawning used to be in 16.0 and 17.1 but was disabled on 2020/12/21 due to breaking VoLTE on select devices.
I have recently ported it to 18.1, but it is still disabled.

**Reply by rsssxd** [reply]
Are there any re-enable plans for 17.1 and 16.1?

**Reply by SkewedZeppelin** [reply]
Only if you want a phone that can't make phone calls.

**Reply by rsssxd** [reply]
If it won't be enabled by default and users can disable it why not?

I don't think it will give the same error on every device.

**Reply by SkewedZeppelin** [reply]
exec-spawning is now in 16.0+
https://divestos.org/index.php?page=news#2022-04
